src/hal/src/acceleration_structure.rs

```diff
@@ -0,0 +1,232 @@
+//! TODO docs
+
+// TODO remove and add docs
+#![allow(missing_docs)]
+
+use std::ops::Range;
+
+use crate::{buffer::Offset, format::Format, Backend, IndexType};
+
+/// Denotes the type of acceleration structure.
+#[derive(Debug)]
+pub enum AccelerationStructureType {
+    ///
+    TopLevel,
+    ///
+    BottomLevel,
+    // TODO vulkan supports "generic level" (where the concrete build type is specified), but discourages its use for applications "written directly for Vulkan" since it "could affect capabilities or performance in the future" (https://www.khronos.org/blog/vulkan-ray-tracing-final-specification-release). Perhaps this is to better support `vkd3d-proton`, but we probably don't want it exposed in gfx?
+}
+
+/// A description of the data needed to build an acceleration structure.
+#[derive(Debug)]
+pub struct AccelerationStructureDesc<'a, B: Backend> {
+    /// TODO: document lifetime required for this buffer
+    pub buffer: &'a B::Buffer,
+    /// TODO: the final gpu address on DX12 needs to be D3D12_RAYTRACING_ACCELERATION_STRUCTURE_BYTE_ALIGNMENT (256), but vulkan only requires this offset to be 256 (as defined by https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VkAccelerationStructureCreateInfoKHR.html)
```

Comment:
`D3D12_RAYTRACING_ACCELERATION_STRUCTURE_BYTE_ALIGNMENT` is a `#define`, is it reasonable to assume this (and the 256 from the Vulkan spec) is a "forever constant" that we can inherit as a static value in gfx?

Reply:
Sure! Let's just make it a constant here.